sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -870,7 +874,8 @@ class Analyzer(
         case SessionCatalogAndIdentifier(catalog, ident) =>
           CatalogV2Util.loadTable(catalog, ident).map {
             case v1Table: V1Table =>
-              v1SessionCatalog.getRelation(v1Table.v1Table)
+              AnalysisContext.get.relationCache.getOrElseUpdate(
+                v1Table.v1Table.qualifiedName, v1SessionCatalog.getRelation(v1Table.v1Table))
```

Comment:
the key should be a fully qualified name, including catalog name. We may have different tables in different catalogs with the same name.

how about
```
val key = catalog.name +: ident.namespace :+ ident.name
relationCache.getOrElseUpdate(key, ...)
```
```

```